Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift

```diff
@@ -3,42 +3,46 @@ import CustomAuth
 import UIKit
 
 class GoogleWeb3AuthLoginModel {
-    var onClose: ((_ key: String?, _ email: String?) -> Void)
+    var authorizationComplete: () -> Void
+    var keyGenerationComplete: ((_ key: String, _ email: String?) -> Void)
 
-    init(onClose: @escaping ((_ key: String?, _ email: String?) -> Void)) {
-        self.onClose = onClose
+    init(authorizationComplete: @escaping () -> Void, keyGenerationComplete: @escaping ((_ key: String, _ email: String?) -> Void)) {
+        self.authorizationComplete = authorizationComplete
+        self.keyGenerationComplete = keyGenerationComplete
     }
 
     static func handle(url: URL) -> Bool {
-        LogService.shared.debug("----> handle(url: \(url.absoluteString))")
-        if url.absoluteString.starts(with: App.configuration.web3auth.redirectUrl) {
+        if url.absoluteString.starts(with: App.configuration.web3auth.redirectScheme + "://") {
             CustomAuth.handle(url: url)
             return true
         }
         return false
     }
 
-    func loginWithCustomAuth(caller: UIViewController) {
+    func loginWithCustomAuth() {
         Task {
-            let sub = SubVerifierDetails(loginType: .web,
+            let sub = SubVerifierDetails(loginType: .installed,
                                          loginProvider: .google,
                                          clientId: App.configuration.web3auth.googleClientId,
                                          verifier: App.configuration.web3auth.googleVerifier,
-                                         redirectURL: App.configuration.web3auth.redirectUrl
+                                         redirectURL: App.configuration.web3auth.redirectScheme + "://"
```

Comment:
this is the second place were we append "://", I would prefer if we predefine it